detectron2/engine/train_loop.py

```diff
@@ -325,7 +325,7 @@ def _write_metrics(
 
     @staticmethod
     def write_metrics(
-        loss_dict: Mapping[str, torch.Tensor],
+        loss_dict: Mapping[str, Union[torch.Tensor, float]],
```

Comment:
You can already write metrics from inside the model like this: https://github.com/facebookresearch/detectron2/blob/80e2673da161f57afe37ef769836a61976108ef1/detectron2/modeling/roi_heads/fast_rcnn.py#L114
With that, the trainer does not need to be changed to include "non-loss" in "loss dict".

Reply:
Thanks, this works. But one issue with this is that it will only show item that contains the name "loss", but some of the items we want to print are not actually losses. Do you have a recommend way to support this?

Reply:
For that I think it's good to change MetricsPrinter (e.g. add an argument or hard code "[metrics]"). But no need to change `train_loop.py` then.